dipy/segment/metricspeed.pyx

```diff
@@ -290,6 +303,49 @@ cdef class MinimumAverageDirectFlipMetric(AveragePointwiseEuclideanMetric):
         return min(dist_direct, dist_flipped)
 
 
+cdef class CosineMetric(CythonMetric):
+    r""" Computes the cosine distance between two vectors.
+
+    A vector of N-dimensional points is represented as a 2D array with
+    shape (1, nb_dimensions).
+
+    Notes
+    -----
+    The distance between two vectors $v_1$ and $v_2$ is equal to
+    $\frac{1}{\pi} \arccos\left(\frac{v_1 \cdot v_2}{\|v_1\| \|v_2\|}\right)$
+    and is bounded within $[0,1]$.
+    """
+    def __init__(CosineMetric self, Feature feature):
+        super(CosineMetric, self).__init__(feature=feature)
+
+    cdef int c_are_compatible(CosineMetric self, Shape shape1, Shape shape2) nogil except -1:
+        return same_shape(shape1, shape2) and shape1.dims[0] == 1
```

Comment:
if same_shape(shape1, shape2)!=0 and shape1.dims[0] == 1:
    return 1
else:
    return 0
to avoid Cython trying to create a boolean object:
https://travis-ci.org/nipy/dipy/jobs/58217418#L775